### PR TITLE
Re-arrange Some Requirements Files

### PR DESCRIPTION
**Pull request by tommy-waltmann**
## Description

This PR changes the directory of the oldest requirements file so dependabot will not try to bump the versions anymore.

## Motivation and Context

On #997 , we attempted a fix so dependabot would find our testing requirements files and resume bumping them monthly again. This attempt made it so dependabot also tried to bump the oldest testing requirements file as well, which is undesirable.

## How Has This Been Tested?

Similar to #997 , we cannot test that dependabot doesn't sense the oldest requirements file until the PR is merged, we can only follow other examples. We can, however, observe that the python 3.6 tests pass, meaning that the files path adjustments were done correctly.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [ ] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation (if relevant).
- [ ] I have added tests that cover my changes (if relevant).
- [ ] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

**Review by bdice — APPROVED** [review]
Approving this -- but I'd suggest adopting the structure we _know_ works for signac, rather than making up a new structure that is untested...

**Comment by joaander** [comment]
> Approving this -- but I'd suggest adopting the structure we _know_ works for signac, rather than making up a new structure that is untested...

Agreed. I suggested that Tommy use a separate directory. However, I was thinking of a separate directory *outside* the `env-files` directory (specified as the root in the dependabot config) so that the behavior would not change subject future changes in the way dependabot scans.

I worry that this PR as is won't change anything. *signac*'s config demonstrates that dependabot performs some amount of recursive searching. It is possible that it will search inside `oldest`, though it may not (the behavior is not documented).

**Comment by tommy-waltmann** [comment]
Okay, I can re-arrange some things. I created a different structure previously because I was unaware at the time about the troubles with dependabot search paths.

**Comment by tommy-waltmann** [comment]
Going to wait until this round of dependabot PRs gets merged before moving further with this. Its creating merge conflicts that I only want to go about resolving once after all the dependency versions are bumped.

**Comment by bdice** [comment]
Sorry, bumped the close button. I was trying to comment.

If you update/move the requirements files first, dependabot will close or recreate the corresponding PRs for new versions. You won’t have to fix merge conflicts.

**Comment by tommy-waltmann** [comment]
This appears to be working right as of now. I find it awkward, however, that the conda environment files are not in the same directories as their corresponding pip requirements files due to the new structure that we need for dependabot to work right. I don't know if it would violate common practice or not to put conda env files in a `requirements/` directory.

File: .github/dependabot.yml
Comment (bdice) [inline .github/dependabot.yml:3]: This PR disables dependabot for pip packages entirely. That seems wrong.

**Review by bdice — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml
Comment (tommy-waltmann) [reply .github/dependabot.yml:3]: Yes, that is wrong.

**Review by bdice — APPROVED** [review]

**Review by joaander — COMMENTED** [review]
I have no comments that I have not already provided in other threads.

**Pull request merged**
